lib/std/time.zig

```diff
@@ -33,16 +33,50 @@ pub fn timestamp() u64 {
 /// Get the posix timestamp, UTC, in milliseconds
 /// TODO audit this function. is it possible to return an error?
 pub fn milliTimestamp() u64 {
+    return @divFloor(nanoTimestamp(), millisecond);
+}
+
+const DarwinTimeStart = struct {
+    timebase: os.darwin.mach_timebase_info_data,
+    inittime: os.darwin.timespec,
+    initclock: u64,
+};
+
+var global_timestart: DarwinTimeStart = undefined;
+var init_global_timestart_once = std.once(init_global_timestart);
+
+pub fn init_global_timestart() void {
+    var micro: os.darwin.timeval = undefined;
+    var timestart: DarwinTimeStart = undefined;
+
+    os.darwin.mach_timebase_info(&timestart.timebase);
+
+    const err = os.darwin.gettimeofday(&micro, null);
+    assert(err == 0);
+
+    timestart.initclock = os.darwin.mach_absolute_time();
+    timestart.inittime.tv_sec = micro.tv_sec;
+    timestart.inittime.tv_nsec = micro.tv_usec * 1000;
+
+    global_timestart = timestart;
+}
+
+/// Get the posix timestamp, UTC, in nanoseconds
+///
+/// On windows this only has a granularity of 100 nanoseconds.
```

Comment:
FYI posix has `clock_getres` to indicate the granularity of the timer. We should probably wrap that too and mention it in this comment.